framework/src/executioners/Transient.C

```diff
@@ -386,10 +386,11 @@ Transient::execute()
   if (!_app.halfTransient())
   {
     TIME_SECTION(_final_timer);
-
     _problem.finalizeMultiApps();
-    _problem.outputStep(EXEC_FINAL);
+    _problem.execMultiAppTransfers(EXEC_FINAL, MultiAppTransfer::TO_MULTIAPP);
+    _problem.execMultiAppTransfers(EXEC_FINAL, MultiAppTransfer::FROM_MULTIAPP);
```

Comment:
I'm not convinced that this is the right fix, perhaps it is, but we'll need a bit of documentation at the very least.

In the rest of the Transient Executioner, we execute the Multiapp, which properly wraps up the Transfer around it (Transfer To -> execute -> Transfer From). Here you are digging into the problem and just executing both sets of Transfers but without an execute in the middle. I realize that in #9923. There's a case for only running the Transfers because of the expense of running the Multiapp, but this feels like a hack. Is there really that much cost if the Multiapp is only executed once on final? We normally don't have Multiapps execute on "final" so we should just be able to execute the whole Multiapp here instead, which would be more consistent with the current design. Note: Make sure that finalizeMultiApps() is the last thing called. That's where you do things like output (think PerfGraph) so these lines aren't in the right spot anyway.

Reply:
I agree.